doc/source/reference/concepts.md

```diff
@@ -0,0 +1,98 @@
+# Seldon Core Concepts
+
+This page is a work in progress to provide concepts related to Seldon Core.
+
+This work is ongoing and we welcome feedback
+
+## Workflow
+#### Conceptual overview of workflows
+A workflow groups the components of a machine learning system into a logical pipeline. Sets of related components are grouped into graphs. The workflow contains the configuration of the components and the definitions of the inputs and outputs of the system, and of each component.
+
+## Component
+#### Conceptual overview of components
+A component is one of:
+-   Model
+    
+-   Router
+    
+-   Combiner
+    
+-   Transformer
+    
+-   Output Transformer
+
+A program that performs one step in the workflow.
+
+## Machine Learning Deployment
+#### Conceptual overview of machine learning deployments
+A machine learning deployment refers to a predictive model in the Seldon ecosystem of a type associated with Seldon (Seldon Deployments) or Kubeflow (Inference Services).
```

Comment:
An ML deployment could also encompass an inference graph (i.e. not just a predictive model). Seldon Core only focuses on Seldon Deployments, so maybe we could remove the last bit?